### ISSUE TITLE
Read_only buffers for weights (inference)

### ISSUE THREAD

**Issue by abialas1** [plaidml/plaidml#1383]
Use read_only buffer for weights where possible (inference).
System level, change entire stack up the EDSL/keras API.

**Comment by flaub** [comment]
I've begun making changes to the PlaidML APIs to provide a way for users to specify constant weights. I plan to have 2 upcoming PRs:
* Changes to APIs
* Changes to implementation of constants to enable optimizations like batch normalization folding.

**Comment by adstraw** [comment]
Assigning to Jeremy for init / fini work
